src/sentry/sentry_metrics/indexer/id_generator.py

```diff
@@ -0,0 +1,36 @@
+import random
+import time
+
+_VERSION_BITS = 4
+_TS_BITS = 32
+_RANDOM_BITS = 28
+_TOTAL_BITS = _VERSION_BITS + _TS_BITS + _RANDOM_BITS
+assert _TOTAL_BITS == 64
+
+_VERSION = 3
+
+# 1st January 2022
+_EPOCH_START = 1641024000
+
+_RANDOM_MAX = pow(2, _RANDOM_BITS) - 1  # inclusive
+
+
+def get_id() -> int:
+    """
+    Generates IDs for use by indexer storages that do not have autoincrement sequences (e.g. CloudSpanner).
+    This function doesn't not provide any guarantee of uniqueness, just a low probability of collisions.
+    It relies on the database to be strongly consistent and reject writes with duplicate IDs. These should
+    be retried with a newly generated ID.
+
+    The ID generated is in roughly incrementing order.
+    """
+
+    now = int(time.time())
+    time_since_epoch = now - _EPOCH_START
+    rand = random.randint(0, _RANDOM_MAX)
+
+    id = _VERSION << (_TOTAL_BITS - _VERSION_BITS)
```

Comment:
Please invert the bits of the VERSION field.
There will be plenty of space between the version and the first 1 in the timestamp for a very long time.
If we invert the bits we keep the opportunity to increase the version bits if we need as the version number would grow towards the right instead of towards the left:
```
0
1
01
11
001
101
...
```

Reply:
updated